pkg/kubelet/rkt/rkt.go

```diff
@@ -920,10 +927,14 @@ func (r *Runtime) generateRunCommand(pod *api.Pod, uuid, netnsName string) (stri
 		runPrepared = append(runPrepared, "--no-overlay=true")
 	}
 
-	// Network namespace set up in kubelet; rkt networking not used
-	runPrepared = append(runPrepared, "--net=host")
+	// Apply '--net=host' to pod that is running on host network or inside a network namespace.
+	if kubecontainer.IsHostNetworkPod(pod) || r.networkPlugin.Name() != network.DefaultPluginName {
```

Comment:
@dcbw So there are 4 places:
```
kubecontainer.IsHostNetworkPod(pod) || r.networkPlugin.Name() != network.DefaultPluginName 

!kubecontainer.IsHostNetworkPod(pod) && r.networkPlugin.Name() != network.DefaultPluginName

kubecontainer.IsHostNetworkPod(pod) || r.networkPlugin.Name() == network.DefaultPluginName

kubecontainer.IsHostNetworkPod(pod) || r.networkPlugin.Name() == network.DefaultPluginName (same as above)

```

Not exactly the same or opposite. So I intend to keep there as they are.